### PR TITLE
Add environment variable for orchestrator service account

### PR DESCRIPTION
**Pull request by andmosourceability**
## What
This PR adds a new environment variable, CONTAINER_ORCHESTRATOR_SERVICE_ACCOUNT, which allows the user to specify a Kubernetes service account for the container orchestrator. Previously, the value was hard-coded to 'airbyte-admin', which does not work for users who are restricted to specific service accounts.

## How
I tried to follow a similar format to how CONTAINER_ORCHESTRATOR_IMAGE was added. I added a field to ContainerOrchestratorConfig and then extracted the value in AsyncOrchestratorPodProcess when the pod is created. The default value is 'airbyte-admin', so Airbyte users who do not need this feature should be unaffected.

## Recommended reading order
1. `airbyte-commons-worker/src/main/java/io/airbyte/workers/process/AsyncOrchestratorPodProcess.java`
2. `airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java`
3. `airbyte-commons-worker/src/main/java/io/airbyte/workers/sync/LauncherWorker.java`
4. `airbyte-commons-worker/src/main/java/io/airbyte/workers/ContainerOrchestratorConfig.java`
5. The rest 

## 🚨 User Impact 🚨
No, Airbyte users who do not need this feature should be unaffected.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

Output of build (not sure if you need this?): 
`BUILD SUCCESSFUL in 4m 47s
449 actionable tasks: 178 executed, 250 from cache, 21 up-to-date`

## Tests

I am not sure if there needs to be a test added for this. It is my first time contributing. Any advice here would be great.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@jdpgrailsdev and @colesnodgrass can you add this to next sprint to review?

**Comment by marcosmarxm** [comment]
Thanks for the contribution @andmosourceability !!!

**Review by jdpgrailsdev — COMMENTED** [review]
The default value (`airbyte-admin`) should be provided on the right hand side of the colon.

**Comment by andmosourceability** [comment]
@jdpgrailsdev Done.

**Comment by andmosourceability** [comment]
Should the default value be in `values.yaml` or `values.yaml.test`?

**Comment by jdpgrailsdev** [comment]
> Should the default value be in `values.yaml` or `values.yaml.test`?

@andmosourceability You only need to provide the env var in the `values.yaml` file if you want to override the default set in `application.yml`.  Therefore, you should be able to add nothing to the Helm values files, unless you want the default Kubernetes deployment to use a different value than the default one provided in the application.

**Comment by andmosourceability** [comment]
@jdpgrailsdev OK, I will remove my changes to the Helm values files then.

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by marcosmarxm** [comment]
@andmosourceability can you solve the conflict?
@jdpgrailsdev can you review this contribution again?

File: airbyte-workers/src/main/resources/application.yml
Comment (jdpgrailsdev) [inline airbyte-workers/src/main/resources/application.yml:59]: @andmosourceability You don't need the back ticks around `airbyte-admin`, unless you literally want those characters in the default value set for the property if the `CONTAINER_ORCHESTRATOR_SERVICE_ACCOUNT` environment variable is unset.

Put another way, I believe that this should be:

`service-account: ${CONTAINER_ORCHESTRATOR_SERVICE_ACCOUNT:airbyte-admin}`

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Comment by sp-bdanies** [comment]
Hello :) Thank you for your PR ! Any news about it?

**Comment by cgardens** [comment]
Hi @andmosourceability , we have moved our platform to a [new public repo](https://github.com/airbytehq/airbyte-platform). Would you be willing to move your work over there? Once you do, I'll make sure we get it merged. Looks like this was ready to go and then got lost. Feel free to tag me @jdpgrailsdev on the new PR. 

Going to close this PR as the code it is reference has moved.

**Pull request closed without merging**
